test_rules_scala.sh

```diff
@@ -16,13 +16,12 @@ $runner bazel build test/...
 #$runner bazel build "test/... --all_incompatible_changes"
 $runner bazel test test/...
 $runner bazel test third_party/...
-# UnusedDependencyChecker doesn't work with strict_java_deps
-$runner bazel build "--strict_java_deps=ERROR -- test/... -test:UnusedDependencyChecker"
-$runner bazel build "--extra_toolchains=//scala:minimal_direct_source_deps -- test/... -test:UnusedDependencyChecker"
-#$runner bazel build "--strict_java_deps=ERROR --all_incompatible_changes -- test/... -test:UnusedDependencyChecker"
-$runner bazel test "--strict_java_deps=ERROR -- test/... -test:UnusedDependencyChecker"
-$runner bazel test "--extra_toolchains=//scala:minimal_direct_source_deps -- test/... -test:UnusedDependencyChecker"
-$runner bazel build "test_expect_failure/missing_direct_deps/internal_deps/... --strict_java_deps=warn"
+$runner bazel build "--extra_toolchains=//test/toolchains:high_level_transitive_deps_strict_deps_error -- test/..."
+$runner bazel build "--extra_toolchains=//scala:minimal_direct_source_deps -- test/..."
+#$runner bazel build "--extra_toolchains=//test/toolchains:high_level_transitive_deps_strict_deps_error --all_incompatible_changes -- test/..."
+$runner bazel test "--extra_toolchains=//test/toolchains:high_level_transitive_deps_strict_deps_warn -- test/..."
+$runner bazel test "--extra_toolchains=//scala:minimal_direct_source_deps -- test/..."
+$runner bazel build "test_expect_failure/missing_direct_deps/internal_deps/... --strict_java_deps=warn --extra_toolchains=//test/toolchains:high_level_transitive_deps_strict_deps_warn"
```

Comment:
@ittiaz We do need both `strict_java_deps` and the toolchain. Because the toolchain controls strictness/deps of scala code under compile, while strict_java_deps controls the strictness/deps of java code under compile (even if the java code is in scala_library). Due to `scala/private/phases/phase_compile.bzl` still using strict_java_deps for java.

At the time I did do this intentionally but now wonder if it was a good idea given that it sometimes leads to things like this (there's a few other places where we don't change strict_java_deps to toolchain because they were testing strict deps on java source files). Anyways this is the existing behavior at least.

Reply:
I think this is actually important because this means that people who have java code in both scala_library and java_library control them together

Reply:
Ok, fine with me